pkgs/swift2objc/test/unit/filter_test_output_b.swift

```diff
@@ -0,0 +1,177 @@
+// Test preamble text
+
+import Foundation
+
+@objc public class ElectricCarWrapper: NSObject {
+  var wrappedInstance: ElectricCar
+  
+  @objc public var batteryCapacity: Int {
+    get {
+      wrappedInstance.batteryCapacity
+    }
+    set {
+      wrappedInstance.batteryCapacity = newValue
+    }
+  }
+  
+  init(_ wrappedInstance: ElectricCar) {
+    self.wrappedInstance = wrappedInstance
+  }
+  
+  @objc init(make: String, model: String, dimensions: DimensionsWrapper, numberOfDoors: Int, tires: TireWrapper, batteryCapacity: Int) {
+    wrappedInstance = ElectricCar(make: make, model: model, dimensions: dimensions.wrappedInstance, numberOfDoors: numberOfDoors, tires: tires.wrappedInstance, batteryCapacity: batteryCapacity)
+  }
+  
+  @objc public func chargeBattery() {
+    wrappedInstance.chargeBattery()
+  }
+}
+
+@objc public class CarWrapper: NSObject {
+  var wrappedInstance: Car
+  
+  @objc public var numberOfDoors: Int {
+    get {
+      wrappedInstance.numberOfDoors
+    }
+    set {
+      wrappedInstance.numberOfDoors = newValue
+    }
+  }
+  
+  @objc public var tires: TireWrapper {
+    get {
+      TireWrapper(wrappedInstance.tires)
+    }
+    set {
+      wrappedInstance.tires = newValue.wrappedInstance
+    }
+  }
+  
+  init(_ wrappedInstance: Car) {
+    self.wrappedInstance = wrappedInstance
+  }
+  
+  @objc init(make: String, model: String, engine: EngineWrapper, dimensions: DimensionsWrapper, numberOfDoors: Int, tires: TireWrapper) {
+    wrappedInstance = Car(make: make, model: model, engine: engine.wrappedInstance, dimensions: dimensions.wrappedInstance, numberOfDoors: numberOfDoors, tires: tires.wrappedInstance)
+  }
+  
+  @objc public func honk() {
+    wrappedInstance.honk()
+  }
+}
+
+@objc public class GarageWrapper: NSObject {
+  var wrappedInstance: Garage
+  
+  init(_ wrappedInstance: Garage) {
+    self.wrappedInstance = wrappedInstance
+  }
+  
+  @objc override init() {
+    wrappedInstance = Garage()
+  }
+  
+  @objc public func addVehicle(vehicle: VehicleWrapper) {
+    wrappedInstance.addVehicle(vehicle: vehicle.wrappedInstance)
+  }
+  
+  @objc public func listVehicles() {
+    wrappedInstance.listVehicles()
+  }
+}
+
+@objc public class BicycleWrapper: NSObject {
+  var wrappedInstance: Bicycle
+  
+  @objc public var dimensions: DimensionsWrapper {
+    get {
+      DimensionsWrapper(wrappedInstance.dimensions)
+    }
+    set {
+      wrappedInstance.dimensions = newValue.wrappedInstance
+    }
+  }
+  
+  @objc public var brand: String {
+    get {
+      wrappedInstance.brand
+    }
+    set {
+      wrappedInstance.brand = newValue
+    }
+  }
+  
+  @objc public var gearCount: Int {
+    get {
+      wrappedInstance.gearCount
+    }
+    set {
+      wrappedInstance.gearCount = newValue
+    }
+  }
+  
+  init(_ wrappedInstance: Bicycle) {
+    self.wrappedInstance = wrappedInstance
+  }
+  
+  @objc init(brand: String, gearCount: Int, dimensions: DimensionsWrapper) {
```

Comment:
Yeah, I thought this might happen. This code won't compile, because you're not generating `DimensionsWrapper`. This is what makes filtering tricky, because you need to think about transitive dependencies.

There are a few ways to handle it:
1. Omit any methods/properties/initializers that refer to declarations that aren't included. I think this would be confusing to users, and not expected behavior.
2. Replace any omitted declarations with `NSObject`, and just cast them to the target object type instead of doing `.wrappedInstance`. This is easy, but loses type safety. Also I don't think it'll work for structs.
3. Fully generate all transitive deps. This can and up generating quite a lot of bindings the user doesn't care about, because of the *transitive* bit. You have to generate the deps of the deps too.
4. Generate the direct deps of the included declarations as stubs (a stub would be like a class that only has the `wrappedInstance` and one wrapping `init` method, no other properties or methods). This is the best solution, because it avoids the need to recursively generate deps of deps (the deps are stubs, so they don't have any deps). It's harder to implement though, because you have to upgrade the code generator to have a notion of stub generation.

Ffigen used to do (3), but I recently upgraded it to do (4) by default (with config options to do (3) if you want). Jnigen used to do (2), but plans to switch to (4).

(4) is a bit hard to implement though, so feel free to do (3) for now.

Reply:
The current solution uses (3). 
I can work on (4) as an improvement.

Reply:
Don't worry about implementing (4) for now. I'm not sure if it's the right approach for swift2objc anyway. This tool is going to be wrapped by swiftgen, and ffigen will be applying its own filters. It might be better for swift2objc's filtering to be less strict than ffigen's. A quick first pass like this to get rid of some of the unnecessary stuff, followed by a strict pass by ffigen sounds reasonable to me. We can do (4) if this approach causes problems.